<html><head>
      <meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
   <title>Appendix A: Old Annotation Server Format</title><meta name="generator" content="DocBook XSL Stylesheets V1.70.1"><link rel="start" href="index.html" title="Cytoscape User Manual"><link rel="up" href="index.html" title="Cytoscape User Manual"><link rel="prev" href="ar01s20.html" title="Acknowledgements"><link rel="next" href="ar01s22.html" title="Appendix B: GNU Lesser General Public License"></head><body bgcolor="white" text="black" link="#0000FF" vlink="#840084" alink="#0000FF"><div class="section" lang="en"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="Appendix A: Old Annotation Server Format"></a>Appendix A: Old Annotation Server Format</h2></div></div></div><p>Handlers for the following format still exist in Cytoscape as legacy code, however we <span class="emphasis"><em><span class="strong"><strong>strongly</strong></span></em></span> recommend using the new formats (OBO + Gene Association) described in the previous section, since they are easier to download directly from the Gene Ontology project and use directly.  Currently, users have no access to an import interface for this old format.  </p><p></p><div class="section" lang="en"><div class="titlepage"><div><div><h3 class="title"><a name="Building your own annotation files"></a>Building your own annotation files</h3></div></div></div><p>The annotation server requires that the gene annotations and associated ontology of controlled vocabulary terms follow a simple format. This simple format was chosen because it is efficient to parse and easy to use. </p><p>The flat file formats are explained below: </p><p></p><div class="section" lang="en"><div class="titlepage"><div><div><h4 class="title"><a name="The Ontology Format"></a>The Ontology Format</h4></div></div></div><p>By example (the Gene Ontology - GO): </p><p></p><pre class="screen">(curator=GO) (type=all)
0003673 = Gene_Ontology
0003674 = molecular_function [partof: 0003673 ]
0008435 = anticoagulant [isa: 0003674 ]
0016172 = antifreeze [isa: 0003674 ]
0016173 = ice nucleation inhibitor [isa: 0016172 ]
0016209 = antioxidant [isa: 0003674 ]
0045174 = glutathione dehydrogenase (ascorbate) [isa: 0009491 0015038 0016209 0016672 ]
0004362 = glutathione reductase (NADPH) [isa: 0015038 0015933 0016209 0016654 ]
0017019 = myosin phosphatase catalyst [partof: 0017018 ]
...
</pre><p>A second example (KEGG pathway ontology): </p><p></p><pre class="screen">(curator=KEGG) (type=Metabolic Pathways)
90001 = Metabolism
80001 = Carbohydrate Metabolism [isa: 90001 ]
80003 = Lipid Metabolism [isa: 90001 ]
80002 = Energy Metabolism [isa: 90001 ]
80004 = Nucleotide Metabolism [isa: 90001 ]
80005 = Amino Acid Metabolism [isa: 90001 ]
80006 = Metabolism of Other Amino Acids [isa: 90001 ]
80007 = Metabolism of Complex Carbohydrates [isa: 90001 ]
...
</pre><p>The format has these required features: </p><div class="itemizedlist"><ul type="disc"><li><p>The first line contains two parenthesized assignments for <code class="code">curator</code> and <code class="code">type</code>. In the GO example above, the ontology file (which is created from the XML that GO provides) nests all three specific ontologies (molecular function, biological process, cellular component) below the 'root' ontology, named 'Gene_Ontology'. <code class="code">(type=all)</code> tells you that all three ontologies are included in that file. </p></li><li>Following the mandatory title line, there are one or more category lines, each with the form: <div class="itemizedlist"><ul type="circle"><li><p><code class="code">number0 = name [isa:|partof: number1 number2 ...]</code> </p></li></ul></div><p>where <code class="code">isa</code> and <code class="code">partof</code> are terms used in GO; they describe the relation between parent and child terms in the ontology hierarchy. </p></li><li><p>The trailing blank before each left square bracket is <span class="emphasis"><em>not</em></span> required; it is an artifact of the python script that creates these files. </p></li></ul></div><p></p></div><div class="section" lang="en"><div class="titlepage"><div><div><h4 class="title"><a name="The Annotation Format"></a>The Annotation Format</h4></div></div></div><p>By example (from the GO biological process annotation file): </p><p></p><pre class="screen">(species=Saccharomyces cerevisiae) (type=Biological Process) (curator=GO)
YMR056C = 0006854
YBR085W = 0006854
YJR155W = 0006081
...
</pre><p>and from KEGG: </p><p></p><pre class="screen">(species=Mycobacterium tuberculosis) (type=Metabolic Pathways) (curator=KEGG)
RV0761C = 10
RV0761C = 71
RV0761C = 120
RV0761C = 350
RV0761C = 561
RV1862 = 10
...
</pre><p>The format has these required features: </p><div class="itemizedlist"><ul type="disc"><li><p>The first line contains three parenthesized assignments, for <code class="code">species</code>, <code class="code">type</code> and <code class="code">curator</code>. In the example just above, the annotation file (created for budding yeast from the flat text file maintained by SGD for the Gene Ontology project and available both at their web site and at GO's) shows three yeast ORFs annotated for biological process with respect to GO, as described above. </p></li><li>Following the mandatory title line, there are one or more annotation lines, each with the form: <div class="itemizedlist"><ul type="circle"><li><p><code class="code">canonicalName = ontologyTermID</code> </p></li></ul></div></li><li><p>Once loaded, this annotation (along with the accompanying ontology) can be assigned to nodes in a Cytoscape network. <span class="strong"><strong>For this to work, the species type of the node must exactly match the species named on the first line of the annotation file. The canonicalName of your node must exactly match the canonicalName present in the annotation file.</strong></span> If you don&#8217;t see the expected results when using this feature of Cytoscape, check this again, as getting either of these wrong is a common mistake. </p></li></ul></div><p></p></div></div><div class="section" lang="en"><div class="titlepage"><div><div><h3 class="title"><a name="Load Data into Cytoscape"></a>Load Data into Cytoscape</h3></div></div></div><p>The easiest way to make annotations available to Cytoscape is by loading annotations into the Cytoscape annotation server. This is the default behavior for the official release of Cytoscape. </p><p></p><div class="section" lang="en"><div class="titlepage"><div><div><h4 class="title"><a name="The Annotation Manifest"></a>The Annotation Manifest</h4></div></div></div><p>You must first create a text file to specify the files you want Cytoscape to load. Here is an example, from a file which (for convenience) we usually call <code class="code">manifest</code>: </p><p></p><pre class="screen">ontology=goOntology.txt
annotation=yeastBiologicalProcess.txt
annotation=yeastMolecularFunction.txt
annotation=yeastCellularComponent.txt
</pre><p>Use the Cytoscape <code class="code">-b</code> command line argument to specify the annotation manifest file to read (e.g. <code class="code">-b manifest</code>). Please note that the <code class="code">-s</code> switch, which sets the default species for your data, is required to exactly match the species named in any annotation file you wish to use. </p><p></p></div></div><div class="section" lang="en"><div class="titlepage"><div><div><h3 class="title"><a name="Getting and Reformatting GO Data"></a>Getting and Reformatting GO Data</h3></div></div></div><p>The Gene Ontology (GO) project is a valuable source of annotation for the genes of many organisms. In this section we will explain how to: </p><div class="orderedlist"><ol type="1"><li>Obtain the GO ontology file </li><li>Reformat it into the simpler flat file Cytoscape uses </li><li>Obtain an annotation file (we illustrate with yeast and human annotation) </li><li>Reformat the annotation files into the simple Cytoscape format </li></ol></div><p></p><div class="section" lang="en"><div class="titlepage"><div><div><h4 class="title"><a name="Obtain the GO ontology file"></a>Obtain the GO ontology file</h4></div></div></div><p>Go to the GO XML FTP (<a href="ftp://ftp.geneontology.org/pub/go/xml/" target="_top">ftp://ftp.geneontology.org/pub/go/xml/</a>) page. Download the latest <code class="code">go-YYYYMM-termdb.xml.gz</code> file. </p><p></p></div><div class="section" lang="en"><div class="titlepage"><div><div><h4 class="title"><a name="Reformat GO XML ontology file into a flat file"></a>Reformat GO XML ontology file into a flat file</h4></div></div></div><p></p><pre class="screen"> gunzip go-YYYYMM-termdb.xml.gz
 python parseGoTermsToFlatFile.py go-YYYYMM-termdb.xml &gt; goOntology.txt
</pre><p>(see below for Python script listing) </p><p></p></div><div class="section" lang="en"><div class="titlepage"><div><div><h4 class="title"><a name="Obtain the association file for your organism"></a>Obtain the association file for your organism</h4></div></div></div><p>GO maintains a list of association files for many organisms; these files associate genes with GO terms. The next step is to get the file for the organism(s) you are interested in, and parse it into the form Cytoscape needs. A list of files may be seen at <a href="http://www.geneontology.org/GO.current.annotations.shtml" target="_top">http://www.geneontology.org/GO.current.annotations.shtml</a>. The rightmost column contains links to tab-delimited files of gene associations, by species. Choose the species you are interested in, and click 'Download'. </p><p>Let's use "GO Annotations @ EBI: Human" as an example. After you have downloaded and saved the file, look at the first few lines: </p><p></p><pre class="screen">SPTR    O00115  DRN2_HUMAN              GO:0003677      PUBMED:9714827  TAS             F       Deoxyribonuclease II precursor  IPI00010348     protein taxon:9606              SPTR
SPTR    O00115  DRN2_HUMAN              GO:0004519      GOA:spkw        IEA             F       Deoxyribonuclease II precursor  IPI00010348     protein taxon:9606        20020425      SPTR
SPTR    O00115  DRN2_HUMAN              GO:0004531      PUBMED:9714827  TAS             F       Deoxyribonuclease II precursor  IPI00010348     protein taxon:9606              SPTR
...
</pre><p>Note that line wrapping has occurred here, so each line of the actual file is wrapped to two lines. The goal is to create from these lines the following lines: </p><p></p><pre class="screen">(species=Homo sapiens) (type=Molecular Function) (curator=GO)
IPI00010348 = 0003677
IPI00010348 = 0004519
IPI00010348 = 0004531
...
</pre><p>or </p><p></p><pre class="screen">(species=Homo sapiens) (type=Biological Process) (curator=GO)
NP_001366 = 0006259
NP_001366 = 0006915
NP_005289 = 0007186
NP_647593 = 0006899
...
</pre><p>The first sample contains molecular function annotations for proteins, and each protein is identified by its IPI number. IPI is the International Protein Index, which maintains cross references to the main databases for human, mouse and rat proteomes.  The second sample contains biological process annotation, and each protein is identified by its NP (RefSeq) number.  These two naming systems, IPI and RefSeq, are two of many that you can use to define canonical names when you run Cytoscape. For budding yeast, it is much easier: the yeast community always uses standard ORF names, and so Cytoscape uses these as canonical names. For human proteins and genes, there is no single standard. </p><p>The solution (for those working with human genes or proteins) is, once you have downloaded the annotations file, to: </p><div class="orderedlist"><ol type="1"><li>Decide which naming system you want to use. </li><li><p>Download <a href="ftp://ftp.ebi.ac.uk/pub/databases/GO/goa/HUMAN/xrefs.goa" target="_top">ftp://ftp.ebi.ac.uk/pub/databases/GO/goa/HUMAN/xrefs.goa</a>. This cross-reference file, when used strategically, allows you to create Cytoscape-compatible annotation files in which the canonical name is the one most meaningful to you. </p></li><li><p>Examine <code class="code">xrefs.goa</code> to figure out which column contains the names you wish to use. </p></li><li>Make a very slight modification to the python script described below, and then </li><li><p>Run that script, supplying both <code class="code">xrefs.goa</code> and that annotation file as arguments. </p></li></ol></div><p>Here are a few sample lines from <code class="code">xrefs.goa</code>: </p><p></p><pre class="screen">SP      O00115  IPI00010348             ENSP00000222219;        NP_001366;              BAA28623;AAC77366;AAC35751;AAC39852;BAB55598;AAB51172;AAH10419; 2960,DNASE2     1777,DNASE2
SP      O00116  IPI00010349             ENSP00000324567;ENSP00000264167;        NP_003650;              CAA70591;       327,AGPS        8540,AGPS
SP      O00124  IPI00010353             ENSP00000265616;ENSP00000322580;        NP_005662;              BAA18958;BAA18959;AAH20694;             7993,D8S2298E
...
</pre><p>Note that line wrapping has occurred here &#8211; each line in this example starts with the letters SP. See the README file for more information (<a href="ftp://ftp.ebi.ac.uk/pub/databases/GO/goa/HUMAN/README" target="_top">ftp://ftp.ebi.ac.uk/pub/databases/GO/goa/HUMAN/README</a>). </p><p>Finally, run the script to create your three annotation files for human proteins: </p><div class="itemizedlist"><ul type="disc"><li><p><code class="code">bioproc.anno</code> (GO biological process annotation) </p></li><li><p><code class="code">molfunc.anno</code> (GO molecular function annotation) </p></li><li><p><code class="code">cellcomp.anno</code> (GO cellular component annotation) </p></li></ul></div><p>using the supplied python script. It may be necessary to modify this script slightly if RefSeq identifiers are not used as canonical names or if you are using a more recent version of Python. </p><p></p><pre class="screen">python parseAssignmentsToFlatFileFromGoaProject.py gene_association.goa_human xrefs.goa
</pre><p>(See below for Python script listing) </p><p></p></div></div><div class="section" lang="en"><div class="titlepage"><div><div><h3 class="title"><a name="Python script examples"></a>Python script examples</h3></div></div></div><p>These scripts, as described above, require Python version 2.2 or later. </p><p>Script 1 - parseGoTermsToFlatFile.py </p><p></p><pre class="screen">
# parseGoTermToFlatFile.py:  translate a GO XML ontology file into a simpler
#  Cytoscape flat file
#-----------------------------------------------------------------------------------
# RCS: $Revision: 1.3 $   $Date: 2003/05/18 00:38:43 $
#-----------------------------------------------------------------------------------
import re, pre, sys
#-----------------------------------------------------------------------------------
def flatFilePrint (id, name, isaIDs, partofIDs):
  isa = ''
  if (len (isaIDs) &gt; 0):
    isa = '[isa: '
    for isaID in isaIDs:
      isa += isaID
      isa += ' '
    isa += ']'
  partof = ''
  if (len (partofIDs) &gt; 0):
    partof = '[partof: '
    for partofID in partofIDs:
      partof += partofID
      partof += ' '
    partof += ']'
  result = '~np~%~/np~s = ~np~%~/np~s ~np~%~/np~s ~np~%~/np~s' ~np~%~/np~ (id, name, isa, partof)
  result = result.strip ()
  if (result == 'isa = isa' or result == 'partof = partof'):
    print &gt;&gt; sys.stderr, 'meaningless term: ~np~%~/np~s' ~np~%~/np~ result
  else:
    print result
#-----------------------------------------------------------------------------------
if (len (sys.argv) != 2):
  print 'usage:  ~np~%~/np~s &lt;someFile.xml&gt;' ~np~%~/np~ sys.argv [0]
  sys.exit ();
inputFilename = sys.argv [1];
print &gt;&gt; sys.stderr,  'reading ~np~%~/np~s...' ~np~%~/np~ inputFilename
text = open (inputFilename).read ()
print &gt;&gt; sys.stderr,  'read ~np~%~/np~d characters' ~np~%~/np~ len (text)
regex = '&lt;go:term .*?&gt;(.*?)&lt;/go:term&gt;';
cregex = pre.compile (regex, re.DOTALL)   # . matches newlines
m = pre.findall (cregex, text)
print &gt;&gt; sys.stderr, 'number of go terms: ~np~%~/np~d' ~np~%~/np~ len (m)
regex2 = '&lt;go:accession&gt;GO:(.*?)&lt;/go:accession&gt;.*?&lt;go:name&gt;(.*?)&lt;/go:name&gt;'
cregex2 = re.compile (regex2, re.DOTALL)
regex3 = '&lt;go:isa\s*rdf:resource="http://www.geneontology.org/go#GO:(.*?)"\s*/&gt;'
cregex3 = re.compile (regex3, re.DOTALL)
regex4 = '&lt;go:part-of\s*rdf:resource="http://www.geneontology.org/go#GO:(.*?)"\s*/&gt;'
cregex4 = re.compile (regex4, re.DOTALL)
goodElements = 0
badElements = 0
print '(curator=GO) (type=all)'
for term in m:
  m2 = re.search (cregex2, term)
  if (m2):
    goodElements += 1;
    id = m2.group (1)
    name = m2.group (2)
    isaIDs = []
    m3 = re.findall (cregex3, term);
    for ref in m3:
      isaIDs.append (ref)
    m4 = re.findall (cregex4, term);
    partofIDs = []
    for ref in m4:
      partofIDs.append (ref)
    flatFilePrint (id, name, isaIDs, partofIDs)
  else:
    badElements += 1;
    print &gt;&gt; sys.stderr, 'no match to m2...'
    print &gt;&gt; sys.stderr, "---------------\n~np~%~/np~s\n------------------" ~np~%~/np~ term
print &gt;&gt; sys.stderr,  'goodElements ~np~%~/np~d' ~np~%~/np~ goodElements
print &gt;&gt; sys.stderr, 'badElements ~np~%~/np~d' ~np~%~/np~ badElements
#--------------------------------------
</pre><p>Script 2 - parseAssignmentsToFlatFileFromGoaProject.py </p><p></p><pre class="screen">
import sys
#-----------------------------------------------------------------------------------
def fixCanonicalName (rawName):
# for instance, trim 'YBR085W|ANC3' to 'YBR085W'
  bar = rawName.find ('|')
  if (bar &lt; 0):
    return rawName
  return rawName [:bar]
#-----------------------------------------------------------------------------------
def fixGoID (rawID):
  bar = rawID.find (':') + 1
  return rawID [bar:]
#-----------------------------------------------------------------------------------
def readGoaXrefFile (filename):
  lines = open (filename).read().split ('\n')
  result = {}
  for line in lines:
    if (len (line) &lt; 10):
      continue
    tokens = line.split ('\t')
    ipi = tokens [2]
    np = tokens [5]
    semicolon = np.find (';')
    if (semicolon &gt;= 0):
      np = np [:semicolon]
    if (len (ipi) &gt; 0 and len (np) &gt; 0):
      result [ipi] = np
  return result
#-----------------------------------------------------------------------------------
if (len (sys.argv) != 3):
  print 'error!  parse   &lt;gene_associations file from GO&gt; &lt;goa xrefs file&gt; '
  sys.exit ()
associationFilename = sys.argv [1];
xrefsFilename = sys.argv [2]
species = 'Homo sapiens'
ipiToNPHash = readGoaXrefFile (xrefsFilename)
tester = 'IPI00099416'
print 'hash size: ~np~%~/np~d' ~np~%~/np~ len (ipiToNPHash)
print 'test map: ~np~%~/np~s -&gt; NP_054861: ~np~%~/np~s ' ~np~%~/np~ (tester, ipiToNPHash [tester])
bioproc = open ('bioproc.txt', 'w')
molfunc = open ('molfunc.txt', 'w')
cellcomp = open ('cellcomp.txt', 'w')
bioproc.write ('(species=~np~%~/np~s) (type=Biological Process) (curator=GO)\n' ~np~%~/np~ species)
molfunc.write ('(species=~np~%~/np~s) (type=Molecular Function) (curator=GO)\n' ~np~%~/np~ species);
cellcomp.write ('(species=~np~%~/np~s) (type=Cellular Component) (curator=GO)\n' ~np~%~/np~ species);
lines=open(associationFilename).read().split('\n')
sys.stderr.write ('found ~np~%~/np~d lines\n' ~np~%~/np~ len (lines))

for line in lines:
  if (line.find ('!') == 0 or len (line) &lt; 2):
    continue
  tokens = line.split ('\t')
  goOntology = tokens [8]
  goIDraw = tokens [4]
  goID = goIDraw.split (':')[1]
  ipiName = fixCanonicalName (tokens [10])
  if (len (ipiName) &lt; 1):
    continue


  if (not ipiToNPHash.has_key (ipiName)):
    continue
  refseqName = ipiToNPHash [ipiName]
  printName = refseqName
  #printName = ipiName
  if (ipiName == tester):
    print '~np~%~/np~s (~np~%~/np~s) has go term ~np~%~/np~s' ~np~%~/np~ (tester, printName, goID)
  if (goOntology == 'C'):
    cellcomp.write ('~np~%~/np~s = ~np~%~/np~s\n' ~np~%~/np~ (printName, goID))
  elif (goOntology == 'P'):
    bioproc.write ('~np~%~/np~s = ~np~%~/np~s\n' ~np~%~/np~ (printName, goID))
  elif (goOntology == 'F'):
    molfunc.write ('~np~%~/np~s = ~np~%~/np~s\n' ~np~%~/np~ (printName, goID))
#-----------------------------------------------------------------------------------
</pre></div></div></body></html>